### ISSUE TITLE
[CLI] Upgrading app throws error when using local chart

### ISSUE THREAD

**Issue by janeczku** [rancher/rancher#15711]
**Rancher versions:**
rancher/server or rancher/rancher:
v2.0.8
rancher/cli:
v2.0.4

Several issues when deploying app from local chart:
1) Chart version (from Chart.yaml) is not recorded in API when doing `install` 
2) `upgrade` command throws error

**Steps to Reproduce:**
1. helm create example
2. grep "version" ./example/Chart.yaml
2. rancher app install ./example example-app
3. rancher app sa example-app
4. rancher app upgrade ./example example-app
FATA[0010] version ./test123 not valid for app

**Results:**
```
$ grep "version" ./example/Chart.yaml
version: 0.1.0

$ rancher app sa example-app
CURRENT   REVISION            CREATED
*         apprevision-96bf6   19 Sep 2018 15:46:11 UTC

CURRENT   VERSION

$ rancher app upgrade ./example example-app
FATA[0001] Not found: ./test123
```

**Comment by andrewunderwood-wf** [comment]
+10 - We are currently limited by this bug

**Comment by jiaqiluo** [comment]
The bug is reproduced on CLI v2.0.4
```
➜  rancher-v2.0.4 ./rancher app as example-app
ID                    NAME          STATE     VERSION
p-ghr4m:example-app   example-app   active 
```
The bug fix is validated on CLI v2.2.0-rc8

```
➜  rancher-v2.2.0-rc8 ./rancher app upgrade example-app ./example
➜  rancher-v2.2.0-rc8 ./rancher app as
ID                    NAME          STATE     CATALOG           TEMPLATE   VERSION
p-ghr4m:example-app   example-app   active    local directory   example    0.1.1
```
